src/unix.rs

```diff
@@ -25,6 +25,13 @@ pub struct Acquired {
     byte: u8,
 }
 
+#[derive(Debug)]
+pub enum ErrFromEnv {
+    ParseEnvVar,
+    OpenFile(String),
+    InvalidDescriptor(i32, i32),
```

Comment:
It's really confusing what each field is for and what is the reason for the failure.

If you don't want to use `io::Error`, you should at least include a `reason` on why the fd is rejected: Is it non-existent or is it not a pipe?

```suggestion
    InvalidDescriptor { read: std::os::fd::RawFd, write: std::os::fd::RawFd, reason: &'static str },
```